src/animation/animation_state_instance.cpp

```diff
@@ -6,16 +6,20 @@ using namespace rive;
 AnimationStateInstance::AnimationStateInstance(const AnimationState* state,
                                                ArtboardInstance* instance) :
     StateInstance(state),
-    m_AnimationInstance(state->animation(), instance),
-    m_KeepGoing(true)
-{}
+    // We're careful to always instance a valid animation here as the
+    // StateMachine makes assumptions about AnimationState's producing valid
+    // AnimationStateInstances with backing animations. This was discovered when
+    // using Clang address sanitizer. We previously returned a
+    // SystemStateInstance (basically a no-op StateMachine state) which would
+    // cause bad casts in parts of the code where we assumed AnimationStates
+    // would have create AnimationStateInstances.
+    m_AnimationInstance(state->animationOrEmpty(), instance),
```

Comment:
Just curious why we don't do the null test here instead?

  m_AnimationInstance(state->animation() ? state->animation() : emptyAnimation, instance)

Reply:
We could do that too, thought it'd be slightly more legible and accessible if we needed it elsewhere but that may be thinking too far ahead.